### PR TITLE
CMake presets and other improvements

### PR DESCRIPTION
**Pull request by fealebenpae**
- Add a `CMakePresets.json` file that contains build configurations for all our targeted platforms.
  - presets are optimized for making Debug builds from Visual Studio Code or another CMake-capable IDE
  - the build scripts are updated to build via presets with release optimizations enabled
- Build a dynamic framework for iOS instead of static libraries.
- Remove the set of `dummy()` functions tricking the linker in favor of `/WHOLEARCHIVE`/`-force_load`/`--whole-archive` linker flags forcing the inclusion of all symbols from the C API library.
- Remove the individual iOS and Android platform libraries and include their sources conditionally in the main `realm_dart` target.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,10 @@
-cmake_minimum_required(VERSION 3.19)
+cmake_minimum_required(VERSION 3.21)
```
Comment (blagoev) [inline CMakeLists.txt:1]: we need to mention this in the readme

**Review by blagoev — CHANGES_REQUESTED** [review]

File: flutter/realm_flutter/ios/realm.podspec [hunk 2]
```diff
@@ -25,7 +25,7 @@ Pod::Spec.new do |s|
                                 'src/realm_dart.cpp'
                                 'src/realm_dart_scheduler.cpp'
   s.public_header_files       = 'Classes/**/*.h',
-  s.vendored_frameworks       = 'realm_flutter_ios.xcframework'
+  s.vendored_frameworks       = 'realm_dart.xcframework'
```
[the remark this answers is not in the document]
Comment (nirinchev) [reply flutter/realm_flutter/ios/realm.podspec:28]: We use `realm_dart` everywhere else, so I think it's better to be consistent.

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -10,89 +18,75 @@ set(HEADERS
...
-endif()
+add_library(realm_dart SHARED ${SOURCES} ${HEADERS})
+target_compile_definitions(RealmFFIStatic PUBLIC -DRealm_EXPORTS)
 
-target_include_directories(realm_dart PUBLIC ${INCLUDE_DIRS})
+target_link_libraries(realm_dart dart-dl RealmFFIStatic Realm::Storage)
 
-if(APPLE)
-    target_link_libraries(realm_dart "-framework CoreFoundation")
+# Force the linker to include all symbols from RealmFFIStatic. realm-dart resolves them at runtime, so the linker considers them unused.
```
[the remark this answers is not in the document]
Comment (fealebenpae) [reply src/CMakeLists.txt:26]: It's exposing only the C API functions, not all of Core. I haven't measured the file size difference, but it should be negligible.

File: src/realm.build.use.ccache.cmake [hunk 4]
```diff
@@ -5,8 +5,8 @@ if(CCACHE_PROGRAM)
     # Set up wrapper scripts
     set(C_LAUNCHER   "${CCACHE_PROGRAM}")
     set(CXX_LAUNCHER "${CCACHE_PROGRAM}")
-    configure_file("${CMAKE_BINARY_DIR}/../scripts/launch-c.in" "${CMAKE_BINARY_DIR}/launch-c")
-    configure_file("${CMAKE_BINARY_DIR}/../scripts/launch-cxx.in" "${CMAKE_BINARY_DIR}/launch-cxx")
+    configure_file("${CMAKE_SOURCE_DIR}/scripts/launch-c.in" "${CMAKE_BINARY_DIR}/launch-c")
```
[the remark this answers is not in the document]
Comment (fealebenpae) [reply src/realm.build.use.ccache.cmake:8]: Good point, I will make a merge commit.

**Review by blagoev — APPROVED** [review]
We better mention that ANDROID_NDK path should be special on Windows. Not sure how this will behave on CI but also on other dev's machines will fail with no clear explanation.

**Review by nielsenko — APPROVED** [review]

**Pull request merged**